parsers/EIA.py

```diff
@@ -372,18 +373,24 @@ def fetch_consumption(
     session: Optional[Session] = None,
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
-):
+) -> List[Dict[str, Any]]:
```

Comment:
I'm not all that familiar with python types but would it be possible to define the exact type here as a part of ConsumptionBatch so we can use ConsumptionBatch.type as the return type for example?

This type would be using a set or sets of [TypedDicts](https://peps.python.org/pep-0589/) to get type hinting on the dicts. It would be too much work to do it on a per parser level but since we can reuse these across all parsers I think it will be worth it.

Reply:
The description of the PR is not 100% complete, the goal in the long run would that parsers return those objects. So no need for using a typed dict here, as in the end it will be those objects that will be returned :)

Reply:
That's great!
No need for this then!